tests/network/vmi_networking.go

```diff
@@ -158,6 +158,13 @@ var _ = SIGDescribe("[Serial][rfe_id:694][crit:medium][vendor:cnv-qe@redhat.com]
 			inboundVMIWithCustomMacAddress = tests.WaitUntilVMIReady(inboundVMIWithCustomMacAddress, libnet.WithIPv6(console.LoginToCirros))
 
 			tests.StartTCPServer(inboundVMI, testPort)
+
+			// Wait for inboundVMI to contain a reported interface with assigned IP address
+			Eventually(func() bool {
+				inboundVMI, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Get(inboundVMI.Name, &metav1.GetOptions{})
+				Expect(err).ShouldNot(HaveOccurred())
+				return len(inboundVMI.Status.Interfaces) > 0 && inboundVMI.Status.Interfaces[0].IP != ""
+			}, 180*time.Second, 5*time.Second).Should(BeTrue())
```

Comment:
I would move the body of the eventually to an helper func, something like:

```golang
makeSureVMHasIP := func() bool {
    inboundVMI, err = virtClient.VirtualMachineInstance(tests.NamespaceTestDefault).Get(inboundVMI.Name, &metav1.GetOptions{})
    Expect(err).ShouldNot(HaveOccurred())
    return len(inboundVMI.Status.Interfaces) > 0 && inboundVMI.Status.Interfaces[0].IP != ""
}

Eventually(makeSureVMHasIP, 180*time.Second, 5*time.Second).Should(BeTrue())
```
It doesn't have to be a lambda, this probably could be used in more places.